### ISSUE TITLE
gss codes vs reg codes

### ISSUE THREAD

**Issue by GeoWill** [DemocracyClub/UK-Polling-Stations#2950]
tl;dr:
We need to know which boundary we're storing in the council table, and the corresponding gss code is probably what should be in the UprnToCouncil table.

Longer version from @chris48s in https://github.com/DemocracyClub/UK-Polling-Stations/pull/2703#discussion_r541632577_

- The point of a register code is it identifies an organisation. If the boundary of St Albans Council changes subtly, say from https://mapit.mysociety.org/area/E07000100.html to https://mapit.mysociety.org/area/E07000240.html it is still the same organisation. It keeps the same identifier.
- The point of a GSS code is it identifies a polygon. If the boundary of St Albans Council changes subtly, say from https://mapit.mysociety.org/area/E07000100.html to https://mapit.mysociety.org/area/E07000240.html it is a new polygon. It gets a new identifier.

So a register code is conceptually correct and useful if the information we need to store is "The phone number for St Albans Council is 01727 819294". If the polygon changes, that fact is still true. However, if the information we want to store is "'115 WATLING STREET, AL2 2NN' is in St Albans" (essentially "this point is inside that polygon") it makes more sense to store that information in terms of a GSS code because changing the polygon changes the fact. In moving to a situation where the `uprntocouncil` table stores register codes (not in this PR) we've moved to a situation where we're using register codes to store "this point is inside that polygon" which becomes _super-confusing_ when minor boundary changes happen because it becomes somewhat ambiguous whether the fact in the DB is "115 WATLING STREET is in E07000100" or "115 WATLING STREET is in E07000240". If you're going to try and clean up when to use a register code and when to use a GSS code as a future piece of work, I'd encourage you to think about this principle in deciding when and why we use a particular code for a particular purpose:

- If changing the polygon changes the fact/data point, store the fact against a GSS code
- If changing the polygon doesn't change the fact/data point, store the fact against a register code

At the moment this principle is violated, but I think following that rule probably leads you to using the right code in the right situation which should then be more intuitive and make problems more obvious. This is something we currently do well in Every Election, but poorly in WDIV.

**Comment by GeoWill** [comment]
Closed with #2991
